mrp_multi_level/wizards/mrp_multi_level.py

```diff
@@ -200,45 +211,39 @@ def explode_action(
         mrp_date_demand = mrp_action_date
         if mrp_date_demand < date.today():
             mrp_date_demand = date.today()
-        if not product_mrp_area_id.product_id.bom_ids:
+        bom = self._get_bom_to_explode(product_mrp_area_id)
+        if not bom:
             return False
-        bomcount = 0
-        for bom in product_mrp_area_id.product_id.bom_ids:
-            if not bom.active or not bom.bom_line_ids:
+        for bomline in bom.bom_line_ids.filtered(
+            lambda x: x.product_qty >= 0 or x.product_id.type != "product"
+        ):
+            if self.with_context(mrp_explosion=True)._exclude_from_mrp(
+                bomline.product_id, product_mrp_area_id.mrp_area_id
+            ):
+                # Stop explosion.
                 continue
-            bomcount += 1
-            if bomcount != 1:
+            if bomline._skip_bom_line(product_mrp_area_id.product_id):
                 continue
-            for bomline in bom.bom_line_ids:
-                if bomline.product_qty <= 0.00 or bomline.product_id.type != "product":
-                    continue
```

Comment:
Can you keep this inside the loop instead of applying it with a filtered? with the "for bom in boms.filtered()" you are iterating twice.

Reply:
That's true! I'll go back to the initial version then